update_feed.sh

```diff
@@ -1,26 +1,31 @@
 #!/bin/bash
 set -euo pipefail
-POSTGRES_DSN="$(python3 -c 'import configparser; c = configparser.ConfigParser(); c.read("config.ini"); print(c["gino"]["dsn"])')"
-wget ftp://gtfs.mot.gov.il/israel-public-transportation.zip -O israel-public-transportation.zip
-python3 create_tables.py
-TEMP_DIR="$(mktemp -d)"
-unzip israel-public-transportation.zip -d "$TEMP_DIR"
-pushd "$TEMP_DIR"
-# Hack! use  (U+0007, bell) as a quote character, since this will probably
-# will never appear in the CSV. This will avoid skipping lines when "
-# is used in the original CSV
-echo -n "loading translations: "
-psql $POSTGRES_DSN -c "\\copy translations from 'translations.txt' with csv header quote ''"
-echo -n "loading agencies:     "
-psql $POSTGRES_DSN -c "\\copy agency from 'agency.txt' with csv header"
-echo -n "loading stops:        "
-psql $POSTGRES_DSN -c "\\copy stops from stops.txt with csv header quote ''"
-echo -n "loading routes:       "
-psql $POSTGRES_DSN -c "\\copy routes from routes.txt with csv header quote ''"
-echo -n "loading trips:        "
-psql $POSTGRES_DSN -c "\\copy trips from trips.txt with csv header quote ''"
-echo -n "loading stoptimes:    "
-psql $POSTGRES_DSN -c "\\copy stoptimes from 'stop_times.txt' with csv header"
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
```

Comment:
This looks like a very complicated way to get the path of the script. Can you explain why it is needed?

It might be better to just give the script an optional parameter / env var named `$CURLBUS_DIR` or something, to point to where curlbus is actually installed, and assume `pwd` if the variable is not defined. Then when called externally you could specify that path.

Reply:
I do not know of a simpler way in bash, you are welcome to suggest one.

Reply:
can you explain why this is better than giving the script the path as a parameter? I'm not sure I understand why you added this.

Reply:
Yes,
The reason that you need to run the script of ``update_feeds.sh`` and know where it is located, is because its dependent on the script ``create_tables.py`` which located in the same folder at itself. If you provide a prefix then it would mean you always have to provide a path to the same folder as the scirpt is running. That would be redundant, you will always have to provide the script, and the folder its running from.

Reply:
yuck, bash is disgusting...

Is it possible to configure an env var in the docker container that will always be set for scripts that run in that container? if that was possible, we could set `$CURLBUS_DIR` as an env var, and that would be cleaner and more readable than launching two nested shells.

But it's okay if you don't want to do that.

Reply:
That can be done, but why would you want to do that? It would mean that anyone running outside the container always has to cd in the source folder to run the script - I will do it if you explicitly demand this.

However since I gave you a detailed explanation why I wrote it that way, I would appreciate a counter-example to why not to write it this way. "yuck" on a language that you decided to pick for the script is not a sufficient explanation. Simple is better than complex, is.
Sorry if I am too direct, but I do want to understand where the logic is coming from.

Reply:
because it looks ugly, and I don't want ugly lines in my codebase.

This file has to be in bash because invoking psql from python is silly, and it has to be invoking psql because loading the GTFS db via python is too slow. This "yuck" comment is against *bash*, not against you or your code.

I'm fine with people having to run this script from the directory in which it resides, because it's not like this is the biggest source of friction that makes running curlbus yourself and keeping it up to date.

anyway - as I said it's up to you. I didn't mean to offend or annoy you with my comment. Sorry about that.